### PR TITLE
Copy over MegaMenu component files

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

Part of https://github.com/department-of-veterans-affairs/va.gov-team/issues/16889

The `<MegaMenu>` component isn't part of the design system, so it doesn't need to be in the design system's component library.

The following files are copied over from [here](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/tree/002da23ca4725b00c4c24dd1eced941e7379ba3b/packages/formation-react/src/components/MegaMenu):

- `Column.jsx`
- `MegaMenu.jsx`
- `MegaMenu.unit.spec.jsx`
- `MenuSection.jsx`
- `SubMenu.jsx`
- `arrow-right-blue.jsx`

I've also shifted the important bits from the `.mdx` file into the `/mega-menu/README.md` file, since the document won't exist on the gatsby site once we get rid of the component.

## Testing done

Unit testing, and verified that localhost worked with the mega menu 

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/site-wide/tests/MegaMenu.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { axeCheck } from 'platform/forms-system/test/config/helpers';
```
Comment (bkjohnson) [inline src/platform/site-wide/tests/MegaMenu.unit.spec.jsx:4]: Changing the location of the `axeCheck` import was the only "code change" I made to the migrated megamenu files.

File: src/platform/site-wide/mega-menu/containers/Main.jsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ import recordEvent from '../../../monitoring/record-event';
 import { isLoggedIn } from '../../../user/selectors';
 import { replaceDomainsInData } from '../../../utilities/environment/stagingDomains';
 
-import MegaMenu from '@department-of-veterans-affairs/formation-react/MegaMenu';
+import MegaMenu from '../components/MegaMenu';
```
Comment (bkjohnson) [inline src/platform/site-wide/mega-menu/containers/Main.jsx:15]: Once this PR is merged, I will open another to remove `MegaMenu` from `formation-react`.

**Review by ncksllvn — APPROVED** [review]

File: src/platform/site-wide/mega-menu/README.md [hunk 3]
```diff
@@ -1,17 +1,443 @@
...
 ```
 
-As you can see we pass in a data argument located at `window.VetsGov.headerFooter.megaMenuData`. There is a MetalSmith plugin that we created to add this to the global window. This is done in the build process. You can find the plugin code at */src/site/stages/build/plugins/create-header-footer.js*
+As you can see we pass in a data argument located at `window.VetsGov.headerFooter.megaMenuData`. There is a MetalSmith plugin that we created to add this to the global window. This is done in the build process. You can find the plugin code at _/src/site/stages/build/plugins/create-header-footer.js_
 
 ## Redux
-The MegaMenu uses Redux to store it's state. You can find the container component at */src/platform/site-wide/mega-menu/containers/Main.jsx. This is where all logic for the component lives. You will also find all of the common Redux files (actions, reducers) in the *mega-menu* folder.
+
+The MegaMenu uses Redux to store it's state. You can find the container component at */src/platform/site-wide/mega-menu/containers/Main.jsx. This is where all logic for the component lives. You will also find all of the common Redux files (actions, reducers) in the *mega-menu\* folder.
```
Comment (cvalarida) [inline src/platform/site-wide/mega-menu/README.md:443]:
```suggestion
The MegaMenu uses Redux to store its state. You can find the container component at */src/platform/site-wide/mega-menu/containers/Main.jsx. This is where all logic for the component lives. You will also find all of the common Redux files (actions, reducers) in the *mega-menu\* folder.
```

**Review by cvalarida — APPROVED** [review]
Looks good to me! :see_no_evil:

**Pull request merged**